language/transaction-builder/src/lib.rs

```diff
@@ -227,7 +226,7 @@ pub fn get_transaction_name(code: &[u8]) -> String {
     "<unknown transaction>".to_string()
 }
 
-pub fn allowing_script_hashes() -> Vec<[u8; SCRIPT_HASH_LENGTH]> {
+pub fn allowing_script_hashes() -> BTreeSet<[u8; SCRIPT_HASH_LENGTH]> {
```

Comment:
Why change this? This will be small because it needs to be stored on-chain. It seems reasonable to use a vector everywhere instead of converting back and forth between a set and a vector.